src/actions/SweepDepositERC4626Action.sol

```diff
@@ -0,0 +1,122 @@
+    {}
+
+    /// @inheritdoc IAction
+    function argumentsHash(bytes calldata arguments) public pure returns (bytes32, bytes32) {
+        return (INSTRUCTION_TYPEHASH, hash(abi.decode(arguments, (SweepDepositERC4626))));
+    }
+
+    /// @inheritdoc ISweepDepositERC4626Action
+    function hash(SweepDepositERC4626 memory arguments) public pure returns (bytes32) {
+        return keccak256(
+            abi.encode(
+                ARGUMENTS_TYPEHASH,
+                arguments.vault,
+                arguments.recipient,
+                arguments.threshold,
+                arguments.endBalance,
+                arguments.minTotalAssets,
+                hash(arguments.fee)
+            )
+        );
+    }
+
+    /// @inheritdoc IAction
+    function execute(
+        InstructionLib.Instruction calldata instruction,
+        InstructionLib.Signature calldata,
+        InstructionLib.ExecutionState calldata executionState
+    ) external override returns (bool) {
+        // initial gas measurement for fee calculation
+        uint256 startGas = gasleft();
+
+        // decode the arguments from the instruction
+        SweepDepositERC4626 memory arguments = abi.decode(instruction.arguments, (SweepDepositERC4626));
+
+        // if first execution, validate the input
+        if (executionState.executionCount == 0) {
+            if (
+                arguments.vault == address(0) || arguments.recipient == address(0)
+                    || arguments.endBalance > arguments.threshold || arguments.minTotalAssets == 0
+            ) {
+                revert InvalidArguments();
+            }
+        }
+
+        // get the underlying token
+        address underlyingToken = IERC4626(arguments.vault).asset();
+
+        // get the account's token balance
+        uint256 balance = IERC20(underlyingToken).balanceOf(address(this));
+
+        // if the balance is under the threshold or equal to the endBalance, revert.
+        // the endBalance check is to prevent the instruction from executing
+        // when threshold == endBalance == balance because in this case we would have depositAmount == 0
+        // slither-disable-next-line incorrect-equality
+        if (balance < arguments.threshold || balance == arguments.endBalance) {
+            revert BalanceUnderThreshold();
+        }
+
+        // get the max deposit amount
+        uint256 maxDeposit = IERC4626(arguments.vault).maxDeposit(arguments.recipient);
+
+        // if the max deposit amount is zero, revert
+        if (maxDeposit == 0) {
+            revert MaxDepositZero();
+        }
+
+        // initialize deposit amount
+        uint256 depositAmount = balance - arguments.endBalance;
+
+        // if the max deposit amount is less than the deposit amount,
+        // set the deposit amount to the max deposit amount and emit an event
+        if (depositAmount > maxDeposit) {
+            depositAmount = maxDeposit;
+
+            emit MaxDepositReached(maxDeposit);
```

Comment:
nit: there isn't super complicated math involved here so nbd tho worth adding the actual/recalculated `endBalance` to the event?

Reply:
Nice yeah that seems reasonable, I'll add it